src/ripple/app/reporting/ETLHelpers.h

```diff
@@ -71,6 +71,15 @@ class NetworkValidatedLedgers
         return max_;
     }
 
+    /// Get most recently validated sequence.
+    /// @return sequence of most recently validated ledger, or empty optional
+    /// if no ledgers are known to have been validated.
+    std::optional<uint32_t>
+    tryGetMostRecent()
+    {
+        return max_;
```

Comment:
I think you need to read `max_` with `m_` locked.  Otherwise you can get a torn `optional`.

Reply:
This method should probably be `const` (as should `getMostRecent()` which is probably why this method is not).

Reply:
To make these methods `const`, `std::condition_variable cv_` has to be made `mutable`. I'll make that change too.